### PR TITLE
Calendar

### PR DESCRIPTION
**Pull request by lessuk400**

### CODE REVIEW

File: app/assets/javascripts/script.coffee [hunk 1]
```diff
@@ -0,0 +1,23 @@
+$(document).ready ->
```
Comment (ailecksandr) [inline app/assets/javascripts/script.coffee]: `$ ->`
coffee shortcut

Comment (ailecksandr) [inline app/assets/javascripts/script.coffee:24]: `$ ->`
coffee shortcut

File: app/assets/javascripts/script.coffee [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  $("#js_main").on "change", $("#js_food_item"),  ->
```
Comment (ailecksandr) [inline app/assets/javascripts/script.coffee:25]: `.on 'change', '#js-food-item', ->`

prefer cebab-case here and you can omit jQuery

File: app/assets/javascripts/script.coffee [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      $(".js_food_item_price").last().val($(".js_food_item :selected").last().data("price"));
```
Comment (ailecksandr) [inline app/assets/javascripts/script.coffee:26]: assign complex operations to separate variable or even move to separate method

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -1,2 +1,26 @@
 class ApplicationController < ActionController::Base
+  before_action :configure_permitted_parameters, if: :devise_controller?
```
Comment (ailecksandr) [inline app/controllers/application_controller.rb]: no, I've meant you need to move that to your own `registrations` controller inherited from devise

[source](https://gist.github.com/kinopyo/2343176)

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -1,2 +1,26 @@
...
+    end
+  end
+
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
+
+  private
+
+  def user_not_authorized
+    flash[:warning] = 'You are not admin of this lunch.'
+    redirect_to(request.referrer || root_path)
```
Comment (ailecksandr) [inline app/controllers/application_controller.rb]:
```ruby
redirect_to ..., alert: 'You are not admin of this lunch.'
```
[source](https://apidock.com/rails/ActionController/Base/redirect_to)

and move string to locales
[source](http://guides.rubyonrails.org/i18n.html)

File: app/controllers/application_controller.rb [hunk 6]
```diff
@@ -1,2 +1,26 @@
...
+      u.permit(:username, :email, :password, :name, :password_confirmation, :remember_me, :avatar,
+      :avatar_cache, :remove_avatar)
+    end
+    devise_parameter_sanitizer.permit(:account_update) do |u|
+      u.permit(:username, :email, :password, :name, :password_confirmation, :current_password,
+      :avatar, :avatar_cache, :remove_avatar)
+    end
+  end
+
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
```
Comment (ailecksandr) [inline app/controllers/application_controller.rb]: call macroses right after module `inclusion/extension`

File: app/controllers/application_controller.rb [hunk 7]
```diff
@@ -1,2 +1,26 @@
 class ApplicationController < ActionController::Base
+  before_action :configure_permitted_parameters, if: :devise_controller?
+  include Pundit
+
+  protected
+
+  def configure_permitted_parameters
```
Comment (ailecksandr) [inline app/controllers/application_controller.rb]: you can override separate `sign_up/account_update` methods for this case as far as I remember.
Also move them to your overrided `registrations` controller

Just look here
[source](https://github.com/plataformatec/devise/blob/master/app/controllers/devise/registrations_controller.rb#L139)

File: app/controllers/food_items_controller.rb [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+  def index
+    @food_items = FoodItem.all
+  end
+
+  def new
+    @food_item = FoodItem.new
+  end
+
+  def create
+    @food_item = FoodItem.create(food_items_params)
```
Comment (ailecksandr) [inline app/controllers/food_items_controller.rb:13]: add space between assignation and `render/redirect`

File: app/controllers/menus_controller.rb [hunk 9]
```diff
@@ -0,0 +1,37 @@
+class MenusController < ApplicationController
+  def index
+    @menus = Menu.all
+    authorize @menus
```
Comment (ailecksandr) [inline app/controllers/menus_controller.rb]: `authorize` before action. You have to check first if you have permissions to do something and then act

File: app/controllers/menus_controller.rb [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+    if Date.today.saturday? || Date.today.sunday?
+      flash[:warning] = 'Today is weekends. No need for menu'
+      redirect_to root_path
+    end
+    if Menu.last.date == Date.today
+      flash[:warning] = 'You have already created menu for today.'
+      redirect_to root_path
+    end
+    @facade = Menus::NewFacade.new
+    authorize @facade.menu
```
Comment (ailecksandr) [inline app/controllers/menus_controller.rb]: same in all places

File: app/controllers/menus_controller.rb [hunk 11]
```diff
@@ -0,0 +1,37 @@
+class MenusController < ApplicationController
+  def index
+    @menus = Menu.all
+    authorize @menus
+  end
+
+  def new
+    if Date.today.saturday? || Date.today.sunday?
```
Comment (ailecksandr) [inline app/controllers/menus_controller.rb]: - assign facade above
- define necessary methods in facade
- define all redirect cases with guard clauses, e.g. `redirect_to root_path, alert: t('menus.new.messages.weekend')`

File: app/controllers/menus_controller.rb [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+    if Menu.last.date == Date.today
+      flash[:warning] = 'You have already created menu for today.'
+      redirect_to root_path
+    end
+    @facade = Menus::NewFacade.new
+    authorize @facade.menu
+  end
+
+  def create
+    @menu = Menu.new menus_params
```
Comment (ailecksandr) [inline app/controllers/menus_controller.rb]: parentheses

File: app/controllers/menus_controller.rb [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+    if @menu.save
```
Comment (ailecksandr) [inline app/controllers/menus_controller.rb]: guard clause

[source](https://github.com/bbatsov/ruby-style-guide#no-nested-conditionals)

File: app/controllers/menus_controller.rb [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+    end
+  end
+
+  private
+
+  def menus_params
+    params
+      .require(:menu)
+      .permit(:name, :price, :date,
+      meals_attributes: %i[price food_item_id _destroy])
```
Comment (ailecksandr) [inline app/controllers/menus_controller.rb]: indentation looks weird

FYI: you may find [this source](https://github.com/bbatsov/ruby-style-guide#no-double-indent) useful

File: app/controllers/orders_controller.rb [hunk 15]
```diff
@@ -0,0 +1,23 @@
+class OrdersController < ApplicationController
+  def new
+    @facade = Orders::NewFacade.new
+  end
+
+  def create
+    @order = Order.new order_params
```
Comment (ailecksandr) [inline app/controllers/orders_controller.rb]: parentheses

[source](https://github.com/bbatsov/ruby-style-guide#method-invocation-parens)

File: app/controllers/orders_controller.rb [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+    @order.user_id = current_user.id
```
Comment (ailecksandr) [inline app/controllers/orders_controller.rb]: move all business logic to `operations/orders/create.rb` => `Orders::Create`

*Hint:* `order_params.merge(user_id: user.id)` in those operation will give the necessary params with user included

File: app/controllers/orders_controller.rb [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+    if @order.save
+      redirect_to orders_path
+    else
+      render plain: 'Da zaebalo'
```
Comment (ailecksandr) [inline app/controllers/orders_controller.rb:37]: locales and fix message

File: app/controllers/orders_controller.rb [hunk 18]
```diff
@@ -0,0 +1,23 @@
+class OrdersController < ApplicationController
+  def new
+    @facade = Orders::NewFacade.new
+  end
+
+  def create
+    @order = Order.new order_params
+    @order.user_id = current_user.id
+    if @order.save
```
Comment (ailecksandr) [inline app/controllers/orders_controller.rb]: guard clause

File: app/controllers/orders_controller.rb [hunk 19]
```diff
@@ -0,0 +1,23 @@
...
+      render plain: 'Da zaebalo'
+    end
+  end
+
+  private
+
+  def order_params
+    params
+      .require(:order)
+      .permit(order_meals_attributes: [:meal_id])
```
Comment (ailecksandr) [inline app/controllers/orders_controller.rb]: `%i[meal_id]`

File: app/helpers/menus_helper.rb [hunk 20]
```diff
@@ -0,0 +1,13 @@
...
+  # TODO: read about pundit
+
+  def food_items_for_select(food_items)
+    food_items.map { |item| [item.name, item.id, data: { price: item.price }] }
+  end
+
+  def user_role_status current_user
+    return 'You are admin of menu' if !current_user.nil? && current_user.has_role?(:admin)
+    return 'Hello dear custommer' unless current_user.nil?
+    'Hello. You should log in in our system'
```
Comment (ailecksandr) [inline app/helpers/menus_helper.rb]: add space after guard clauses

File: app/helpers/menus_helper.rb [hunk 21]
```diff
@@ -0,0 +1,13 @@
+module MenusHelper
+  # TODO: read about pundit
+
+  def food_items_for_select(food_items)
+    food_items.map { |item| [item.name, item.id, data: { price: item.price }] }
+  end
+
+  def user_role_status current_user
+    return 'You are admin of menu' if !current_user.nil? && current_user.has_role?(:admin)
```
Comment (ailecksandr) [inline app/helpers/menus_helper.rb]: `current_user&.has_role?(:admin)`

[source](http://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/)

+ locales

File: app/helpers/menus_helper.rb [hunk 22]
```diff
@@ -0,0 +1,13 @@
+module MenusHelper
+  # TODO: read about pundit
+
+  def food_items_for_select(food_items)
+    food_items.map { |item| [item.name, item.id, data: { price: item.price }] }
+  end
+
+  def user_role_status current_user
```
Comment (ailecksandr) [inline app/helpers/menus_helper.rb]: parentheses

[source](https://github.com/bbatsov/ruby-style-guide#method-parens)

File: app/models/menu.rb [hunk 23]
```diff
@@ -0,0 +1,12 @@
+class Menu < ApplicationRecord
+  validates :name, presence: true
+
+  has_many :meals
+  has_many :food_items, through: :meals
+
+  accepts_nested_attributes_for :meals
+
+  def start_time
+    date # #Where 'start' is a attribute of type 'Date' accessible through MyModel's relationship
```
Comment (ailecksandr) [inline app/models/menu.rb:60]: `an attribute` and what actually is that?

File: app/models/user.rb [hunk 24]
```diff
@@ -0,0 +1,20 @@
...
+
+  validates_presence_of :email
+  validates_presence_of :avatar
+  validates_integrity_of :avatar
+  validates_processing_of :avatar
+
+  private
+
+  def set_role
+    add_role :customer
```
Comment (ailecksandr) [inline app/models/user.rb]: what if plain user will not have roles at all? No role - no permissions )

File: app/models/user.rb [hunk 25]
```diff
@@ -0,0 +1,20 @@
...
+         :recoverable, :rememberable, :trackable, :validatable
+
+  validates_presence_of :email
+  validates_presence_of :avatar
+  validates_integrity_of :avatar
+  validates_processing_of :avatar
+
+  private
+
+  def set_role
```
Comment (ailecksandr) [inline app/models/user.rb]: name it like `set_admin!`

File: app/models/user.rb [hunk 26]
```diff
@@ -0,0 +1,20 @@
+class User < ApplicationRecord
+  rolify
+  before_create :set_role
```
Comment (ailecksandr) [inline app/models/user.rb]: `before_create :set_admin!, unless: -> { self.class.exists? }`

File: app/models/user.rb [hunk 27]
```diff
@@ -0,0 +1,20 @@
...
+  validates_presence_of :email
+  validates_presence_of :avatar
+  validates_integrity_of :avatar
+  validates_processing_of :avatar
+
+  private
+
+  def set_role
+    add_role :customer
+    add_role :admin unless self.class.exists?
```
Comment (ailecksandr) [inline app/models/user.rb]: condition can be omitted

File: app/views/menus/new.html.erb [hunk 28]
```diff
@@ -0,0 +1,48 @@
+<div class="container-fluid">
+  <div class="row">
+    <div class="col-4 ">
+          <div class="card-deck mb-3 text-center">
+            <div class="card mb-4 box-shadow" id="js_main">
+              <div class="card-header">
+                <h4 class="my-0 font-weight-normal">Menu creator</h4>
+              </div>
+              <%= simple_form_for @facade.menu, html: { id: "myForm" } do |f| %>
```
Comment (ailecksandr) [inline app/views/menus/new.html.erb]: be conventional. Go with `camelCase` id/classes namings or `kebab-case`. But not `snake_case`

File: app/views/menus/new.html.erb [hunk 29]
```diff
@@ -0,0 +1,48 @@
...
+                <div id="meals">
+                    <%= f.fields_for :meals do |mf| %>
+                      <%= render 'meal_fields', f: mf %>
+                    <% end %>
+
+                    <div class="links">
+                      <%= link_to_add_association 'add meal', f, :meals %>
+                    </div>
+                  </div>
+                  <%= f.button :button, :id => "button_to_add_menu_item", type: "button" do %>
```
Comment (ailecksandr) [inline app/views/menus/new.html.erb]: `..., id: 'blah-blah'`

File: app/views/menus/new.html.erb [hunk 30]
```diff
@@ -0,0 +1,48 @@
...
+                <% end %>
+              </div>
+            </div>
+        </div>
+    </div>
+    <div class="col-6 w-100">
+      <div style="position:fixed">
+        <%= f.button :submit %>
+        <% end %>
+      <% FoodItem.meal_types do |key,value| %>
```
Comment (ailecksandr) [inline app/views/menus/new.html.erb]: `|key, value|` - space after comma

File: app/views/orders/index.html.erb [hunk 31]
```diff
@@ -0,0 +1,8 @@
+<% Order.all.each do |order| %>
+<hr />
+  <% order.order_meals.each do |om| %>
```
Comment (ailecksandr) [inline app/views/orders/index.html.erb]: name it normally, not `om`

File: config/initializers/devise.rb [hunk 32]
```diff
@@ -280,4 +281,6 @@
   # When using OmniAuth, Devise cannot automatically set OmniAuth path,
   # so you need to do it manually. For the users scope, it would be:
   # config.omniauth_path_prefix = '/my_engine/users/auth'
+  config.secret_key = '0a37102ea08449d2150786350d49ccac0a135ddd'\
```
Comment (ailecksandr) [inline config/initializers/devise.rb]: Use environment variables. You can try gem `dotenv` for this. That's very sensitive key

File: elect [hunk 33]
```diff
@@ -0,0 +1,21 @@
+=> [[32m#<Meal:0x00000000058ce808[0m
```
Comment (ailecksandr) [inline elect]: what a heck is that? )

move `public/uploads` => `.gitinore` to avoid adding your uploaded test files to repository

**Review by ailecksandr — CHANGES_REQUESTED** [review]
Keep going

File: .env [hunk 34]
```diff
@@ -0,0 +1 @@
+SECRET_KEY = 0a37102ea08449d2150786350d49ccac0a135dddde5dd7e2999b5e2681d1a61a58fe3c3b444980099272c6f3da504f3a7e3fee0df07dd0f0d90c85e1e6440613
```
Comment (ailecksandr) [inline .env:1]: add file to `.gitignore`

File: app/assets/javascripts/script.coffee [hunk 35]
```diff
@@ -0,0 +1,11 @@
+$ ->
```
Comment (ailecksandr) [inline app/assets/javascripts/script.coffee]: you can even wrap that all in class. That's one of main advantages of Coffeescript

it'll be +- something like:
```
class OrderForm
  FOOD_ITEM_SELECTOR = '.js-food-item'

  constructor: (form) ->
    @form = $(form)

  init: ->
    @bindEvents()

  bindEvents: ->
    @form.on 'change', @FOOD_ITEM_SELECTOR, ->
      let price = $("#{@FOOD_ITEM_SELECTOR}:selected").data('price')

      this.attr('min', price).val(price)

$ ->
  new OrderForm('#js_main').init()

```
[source](https://coffeescript.org)

File: app/assets/stylesheets/hello.scss [hunk 36]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Hello controller here.
```
Comment (ailecksandr) [inline app/assets/stylesheets/hello.scss]: remove unnecessary `scss` files

File: app/controllers/application_controller.rb [hunk 37]
```diff
@@ -1,2 +1,11 @@
 class ApplicationController < ActionController::Base
+  include Pundit
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
```
Comment (ailecksandr) [inline app/controllers/application_controller.rb]: space between macros and module inslusion

File: app/controllers/menus_controller.rb [hunk 38]
```diff
@@ -0,0 +1,32 @@
+class MenusController < ApplicationController
+  def index
+    @menus = Menu.all
+  end
+
+  def new
+    authorize Menus::NewFacade.new.menu
```
Comment (ailecksandr) [inline app/controllers/menus_controller.rb]: `authorize Menu`

File: app/controllers/menus_controller.rb [hunk 39]
```diff
@@ -0,0 +1,32 @@
+class MenusController < ApplicationController
+  def index
+    @menus = Menu.all
```
Comment (ailecksandr) [inline app/controllers/menus_controller.rb]: `authorize Menu`

File: app/controllers/menus_controller.rb [hunk 40]
```diff
@@ -0,0 +1,32 @@
...
+  end
+
+  def new
+    authorize Menus::NewFacade.new.menu
+    @facade = Menus::NewFacade.new
+    redirect_to root_path, flash: {warning: t('menus_messages.new.messages.weekends')} if
```
Comment (ailecksandr) [inline app/controllers/menus_controller.rb]: space above

move `redirect_to root_path, flash ...` to private methods and call them here. Two-line ifs look messy

File: app/controllers/menus_controller.rb [hunk 41]
```diff
@@ -0,0 +1,32 @@
...
+                                                                                @facade.weekend?
+
+    redirect_to root_path, flash: {warning: t('menus_messages.new.messages.already_created')} if
+                                                                      @facade.menu_already_exist?
+  end
+
+  def create
+    @menu = Menu.new(menus_params)
+    redirect_to menus_path and return if @menu.save
```
Comment (ailecksandr) [inline app/controllers/menus_controller.rb:85]: space below & above

`return redirect_to menus_path if @menu.save`

File: app/controllers/orders_controller.rb [hunk 42]
```diff
@@ -0,0 +1,20 @@
+class OrdersController < ApplicationController
+  def index; end
+
+  def new
+    @facade = Orders::NewFacade.new
+  end
+
+  def create
+    @order = Orders::Create.call(current_user.id, order_params)
+    redirect_to orders_path and return if @order.save
```
Comment (ailecksandr) [inline app/controllers/orders_controller.rb:13]: space below & above

`return redirect_to ...`

File: app/controllers/orders_controller.rb [hunk 43]
```diff
@@ -0,0 +1,20 @@
...
+    render plain: 'Something gone wrong'
+  end
+
+  private
+
+  def order_params
+    params
```
Comment (ailecksandr) [inline app/controllers/orders_controller.rb]: you can put it all in one line here

File: app/controllers/users/registrations_controller.rb [hunk 44]
```diff
@@ -0,0 +1,16 @@
+class Users::RegistrationsController < Devise::RegistrationsController
+  before_action :configure_permitted_parameters, if: :devise_controller?
+
+  protected
+
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up) do |u|
```
Comment (ailecksandr) [inline app/controllers/users/registrations_controller.rb:7]: move these two configurations to separate methods and call here

File: app/decorators/order_meal_decorator.rb [hunk 45]
```diff
@@ -0,0 +1,9 @@
+class OrderMealDecorator < ApplicationDecorator
+  delegate :meal_type,        to: :food_item, allow_nil: true
+  delegate :food_item, :menu, to: :meal,      allow_nil: true
+
+  def related_food_items
+    menu.meals.joins(:food_item).where(food_items: { meal_type: meal_type })
+      .map { |obj| [obj.food_item.name, obj.id] }
```
Comment (ailecksandr) [inline app/decorators/order_meal_decorator.rb]: `pluck('food_items.name, meals.id')` instead of `map`

File: app/helpers/menus_helper.rb [hunk 46]
```diff
@@ -0,0 +1,13 @@
+module MenusHelper
+  # TODO: read about pundit
```
Comment (ailecksandr) [inline app/helpers/menus_helper.rb]: You've already read :)

File: app/helpers/menus_helper.rb [hunk 47]
```diff
@@ -0,0 +1,13 @@
...
+
+  def food_items_for_select(food_items)
+    food_items.map { |item| [item.name, item.id, data: { price: item.price }] }
+  end
+
+  def user_role_status(current_user)
+    return t('menus_messages.new.helpers.admin') if current_user&.has_role? :admin
```
Comment (ailecksandr) [inline app/helpers/menus_helper.rb]:
```ruby
# app/models/user.rb

class User
  ...

  def admin?
    has_role?(:admin)
  end
end
```

and `current_user.admin?` here

File: app/helpers/menus_helper.rb [hunk 48]
```diff
@@ -0,0 +1,13 @@
...
+
+    return t('menus_messages.new.helpers.customer') unless current_user.nil?
```
Comment (ailecksandr) [inline app/helpers/menus_helper.rb:42]: `if current_user`

File: app/models/menu.rb [hunk 49]
```diff
@@ -0,0 +1,12 @@
+class Menu < ApplicationRecord
+  validates :name, presence: true
+
+  has_many :meals, inverse_of: :menu
+  has_many :food_items, through: :meals
+
+  accepts_nested_attributes_for :meals
+
+  def start_time
```
Comment (ailecksandr) [inline app/models/menu.rb:59]: You should remove this as I remember, isn't it?

File: app/models/role.rb [hunk 50]
```diff
@@ -0,0 +1,18 @@
...
+              polymorphic: true,
+              optional:    true
+
+  validates :resource_type,
+             inclusion: { in: Rolify.resource_types },
+             allow_nil: true
+
+  scopify
+
+  def is_admin?
```
Comment (ailecksandr) [inline app/models/role.rb]: can be omitted

File: app/policies/menu_policy.rb [hunk 51]
```diff
@@ -0,0 +1,9 @@
+class MenuPolicy < ApplicationPolicy
+  def index?
+    true
+  end
+
+  def new?
+    user&.has_role? :admin
```
Comment (ailecksandr) [inline app/policies/menu_policy.rb]: `user&.admin?`

File: app/views/meals/index.html.erb [hunk 52]
```diff
@@ -0,0 +1,3 @@
+<% @meals.each do |meal| %>
+  <%= puts meal.foodItem.price %>
```
Comment (ailecksandr) [inline app/views/meals/index.html.erb]: remove `puts`

File: elect (hunk 33 above)
Comment (ailecksandr) [reply elect]: remove file

**Review by ailecksandr — CHANGES_REQUESTED** [review]
Excellent work. Small fixes and we're all set to go

Also, please:
- rename commit about "Grobana forma"
- remove all git-ignored files from remote commits

File: app/assets/javascripts/food_items.coffee [hunk 53]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (ailecksandr) [inline app/assets/javascripts/food_items.coffee:1]: remove unnecessary `coffee` files

File: app/assets/javascripts/script.coffee [hunk 54]
```diff
@@ -0,0 +1,20 @@
+class OrderForm
+  FOOD_ITEM_SELECTOR = '.js-food-item-get-price'
+  FOOD_ITEM_DATA = '.js-food-item-set-price'
+  FORM_SELECTOR = '#js-main'
+
+  constructor: (@form) ->
+    @form = $(FORM_SELECTOR)
+    console.log("#{@form}")
```
Comment (ailecksandr) [inline app/assets/javascripts/script.coffee]: `console.log` can be omitted

File: app/controllers/menus_controller.rb [hunk 55]
```diff
@@ -0,0 +1,46 @@
+class MenusController < ApplicationController
+  def index
+    authorize Menu
```
Comment (andrewr224) [inline app/controllers/menus_controller.rb]: Add empty line after `authorize`

File: app/assets/javascripts/script.coffee [hunk 56]
```diff
@@ -0,0 +1,20 @@
+class OrderForm
+  FOOD_ITEM_SELECTOR = '.js-food-item-get-price'
+  FOOD_ITEM_DATA = '.js-food-item-set-price'
+  FORM_SELECTOR = '#js-main'
+
+  constructor: (@form) ->
+    @form = $(FORM_SELECTOR)
```
Comment (ailecksandr) [inline app/assets/javascripts/script.coffee]:
```
constructor: (selector = FORM_SELECTOR) ->
  @form = $(selector)
...
```

File: app/controllers/menus_controller.rb [hunk 57]
```diff
@@ -0,0 +1,46 @@
...
+    params
+      .require(:menu)
+      .permit(:name,
+              :price,
+              :date,
+              meals_attributes: %i[price food_item_id _destroy])
+  end
+
+  def redirect_weekend
+    redirect_to root_path, flash: {
```
Comment (andrewr224) [inline app/controllers/menus_controller.rb:54]: [redirect_to doesn't stop the execution of the function](http://api.rubyonrails.org/classes/ActionController/Redirecting.html)

You need to add `return` to `redirect`

File: app/controllers/menus_controller.rb [hunk 58]
```diff
@@ -0,0 +1,46 @@
...
+      warning: t('menus_messages.new.messages.weekends')
+      } if @facade.weekend?
+  end
+
+  def redirect_menu_exist
+    redirect_to root_path, flash: {
```
Comment (andrewr224) [inline app/controllers/menus_controller.rb:60]: Same as above

File: app/models/food_item.rb [hunk 59]
```diff
@@ -0,0 +1,19 @@
+class FoodItem < ApplicationRecord
+  FIRST_MEAL = 'first_meal'.freeze
+  MAIN_MEAL  = 'main_meal'.freeze
+  DRINK      = 'drink'.freeze
+
+  validates :name, presence: true, uniqueness: {case_sensetive: true}, length: { maximum: 20, minimum: 3 }
+  validates :price, presence: true, numericality: true
+  validates :meal_type, presence: true
+  validates :description, presence: true, length: { maximum: 100, minimum: 10 }
```
Comment (andrewr224) [inline app/models/food_item.rb:14]: Align validations

File: app/policies/application_policy.rb [hunk 60]
```diff
@@ -0,0 +1,53 @@
...
+
+  def new?
+    create?
+  end
+
+  def update?
+    false
+  end
+
+  def edit?
```
Comment (andrewr224) [inline app/policies/application_policy.rb:44]: Make an alias

File: app/policies/application_policy.rb [hunk 61]
```diff
@@ -0,0 +1,53 @@
...
+
+  def show?
+    scope.where(id: record.id).exists?
+  end
+
+  def create?
+    false
+  end
+
+  def new?
```
Comment (andrewr224) [inline app/policies/application_policy.rb:36]: Make an alias

File: app/assets/javascripts/script.coffee [hunk 62]
```diff
@@ -0,0 +1,20 @@
+class OrderForm
```
Comment (ailecksandr) [inline app/assets/javascripts/script.coffee:3]: fix it as has been shown

**Pull request closed without merging**